Source/Core/Core/State.cpp

```diff
@@ -280,8 +281,20 @@ struct CompressAndDumpState_args
 static void CompressAndDumpState(CompressAndDumpState_args save_args)
 {
 	std::lock_guard<std::mutex> lk(*save_args.buffer_mutex);
-	if (!save_args.wait)
+
+	//By creating ScopeGuard instance we ensure that g_compressAndDumpStateSyncEvent.Set()
+	//will be called then we leave this function, even if it's because of exception being thrown.
```

Comment:
We don't use exception handling in Dolphin, so the part about exceptions isn't especially relevant.